site/cds_rdm/inspire_harvester/writer.py

```diff
@@ -132,34 +187,52 @@ def _create_new_version(self, entry, record):
                 if not file_content:
                     return
                 self._create_file(file, file_content, new_version_draft)
+        current_app.logger.info("Files have been successfully created.")
+
         # update metadata TODO make indempotent
+        current_app.logger.info("Start updating metadata.")
         current_rdm_records_service.update_draft(
             system_identity, new_version_draft.id, entry
         )
         current_rdm_records_service.publish(system_identity, new_version_draft.id)
+        current_app.logger.info(
+            "Metadata is successfully updated and record is published."
+        )
 
     def _create_new_record(self, entry):
         """For new records coming from INSPIRE, create and publish a draft in CDS."""
+        current_app.logger.info("Start creating a new record.")
         file_entries = entry["files"].get("entries", None)
         if not file_entries:
-            # TODO log the absence of files
+            current_app.logger.error(
+                f"INSPIRE record #{entry['id']} has no files. Metadata-only records are not supported. Aborting record writer."
+            )
             return
-            # raise WriterError(f"INSPIRE record #{entry['id']} has no files. Aborting.")
-        draft = current_rdm_records_service.create(system_identity, data=entry)
 
+        draft = current_rdm_records_service.create(system_identity, data=entry)
+        current_app.logger.info(f"New draft is created ({draft.id}).")
         try:
+            current_app.logger.info(
+                f"Creating new files for the draft: {file_entries}."
```

Comment:
too detailed to put whoe file metadata here. I would go with either filenames or checksums